crates/nu-cli/src/data/value.rs

```diff
@@ -95,18 +95,26 @@ pub fn compute_values(
                 }?;
                 Ok(UntaggedValue::Primitive(Primitive::Decimal(result)))
             }
-            (Primitive::Date(x), Primitive::Date(y)) => {
-                let result = match operator {
-                    Operator::Minus => Ok(x.signed_duration_since(*y).num_seconds()),
-                    _ => Err((left.type_name(), right.type_name())),
-                }?;
-                Ok(UntaggedValue::Primitive(Primitive::Duration(result)))
-            }
-            (Primitive::Date(x), Primitive::Duration(y)) => {
+            (Primitive::Date(x), Primitive::Date(y)) => match operator {
+                Operator::Minus => Ok(UntaggedValue::Primitive(Primitive::from(
+                    x.signed_duration_since(*y),
+                ))),
+                _ => Err((left.type_name(), right.type_name())),
+            },
+            (Primitive::Date(x), Primitive::Duration(_)) => {
                 let result = match operator {
-                    Operator::Plus => Ok(x
-                        .checked_add_signed(chrono::Duration::seconds(*y as i64))
-                        .expect("Overflowing add of duration")),
+                    Operator::Plus => {
+                        // FIXME: I'm not sure if this is the way the overflows should be handled.
+                        // The previous version used except().
+                        // FIXME: Not sure if I could do something better with the Span.
```

Comment:
`.expect.(...)` is okay for now. We'll probably go back over the code in the future and see if we can fix those, but for now it's okay.